docs/nightly_releases.md

```diff
@@ -117,3 +108,111 @@ pip install -f https://iree.dev/pip-release-links.html --upgrade --pre \
   iree-base-compiler iree-base-runtime --src deps \
   -e "git+https://github.com/iree-org/iree-turbine.git#egg=iree-turbine"
 ```
+
+## Switching between stable and nightly channels
+
+The [`shark-ai` package on PyPI](https://pypi.org/project/shark-ai/) is a
+meta-package that pins specific stable versions of each package that share
+at least their major and minor versions:
+
+```bash
+pip install shark-ai==2.9.1
+
+pip freeze
+# ...
+# iree-base-compiler==2.9.0
+# iree-base-runtime==2.9.0
+# iree-turbine==2.9.0
+# ...
+# shark-ai==2.9.1
+# shortfin==2.9.1
+# ...
+```
+
+If you attempt to update any individual package outside of those supported
+versions, pip will log an error but continue anyway:
+
+```bash
+pip install --upgrade --pre \
+  -f https://github.com/nod-ai/shark-ai/releases/expanded_assets/dev-wheels \
+  shortfin==3.0.0rc20241118
+
+# Looking in links: https://github.com/nod-ai/shark-ai/releases/expanded_assets/dev-wheels
+# Collecting shortfin==3.0.0rc20241118
+#   Downloading https://github.com/nod-ai/shark-ai/releases/download/dev-wheels/shortfin-3.0.0rc20241118-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.5 MB)
+#      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.5/2.5 MB 24.3 MB/s eta 0:00:00
+# Installing collected packages: shortfin
+#   Attempting uninstall: shortfin
+#     Found existing installation: shortfin 2.9.1
+#     Uninstalling shortfin-2.9.1:
+#       Successfully uninstalled shortfin-2.9.1
+# ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
+# shark-ai 2.9.1 requires shortfin==2.9.1, but you have shortfin 3.0.0rc20241118 which is incompatible.
+# Successfully installed shortfin-3.0.0rc20241118
+
+pip freeze
+# ...
+# shark-ai==2.9.1
+# shortfin==3.0.0rc20241118
+# ...
+```
+
+Installing the `shark-ai` package again should get back to aligned versions:
+
+```bash
+pip install shark-ai==2.9.2
```

Comment:
This might be little confusing as one was installing `2.9.1` before but now we install *again* but a different version. Is this intended?